scipy/stats/_qmc.py

```diff
@@ -22,10 +34,30 @@
     _cy_wrapper_update_discrepancy
 )
 
+
 __all__ = ['scale', 'discrepancy', 'update_discrepancy',
            'QMCEngine', 'Sobol', 'Halton', 'LatinHypercube',
            'MultinomialQMC', 'MultivariateNormalQMC']
 
+IntNumber = Union[int, np.integer]
+DecimalNumber = Union[float, np.floating]
+
+# Since Generator was introduced in numpy 1.17, the following condition is needed for
+# backward compatibility
+if TYPE_CHECKING:
+    SeedType = Optional[Union[IntNumber, np.random.Generator,
```

Comment:
I feel like these definitions should be in `utils` as we are likely going to need them everywhere.